Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.mm

```diff
@@ -59,11 +60,226 @@
 #import <WebCore/ResourceResponse.h>
 #import <WebCore/ScrollAnimator.h>
 #import <WebCore/SharedBuffer.h>
+#import <WebCore/WebAccessibilityObjectWrapperMac.h>
 #import <wtf/cocoa/VectorCocoa.h>
 #import <wtf/text/TextStream.h>
 
 #import "PDFKitSoftLink.h"
 
+
+@interface WKAccessibilityPDFDocumentObject: NSObject {
+    RetainPtr<PDFDocument> _pdfDocument;
+    WeakObjCPtr<NSObject> _parent;
+    // FIXME: Should not call pdfPlugin from non-main thread
+    ThreadSafeWeakPtr<WebKit::PDFPluginBase> _pdfPlugin;
+}
+
+@property (assign) WeakPtr<WebCore::HTMLPlugInElement, WebCore::WeakPtrImplWithEventTargetData> pluginElement; 
+
+- (id)initWithPDFDocument:(RetainPtr<PDFDocument>)document andElement:(WebCore::HTMLPlugInElement*)element;
+- (void)setParent:(NSObject *)parent;
+- (void)setPDFDocument:(RetainPtr<PDFDocument>)document;
+- (void)setPDFPluginBase:(WebKit::PDFPluginBase*)plugin;
+- (PDFDocument *)document;
+- (NSObject *)parent;
+- (id)accessibilityHitTest:(NSPoint)point;
+- (void)gotoDestination:(PDFDestination *)destination;
+
+@end
+
+@implementation WKAccessibilityPDFDocumentObject
+
+@synthesize pluginElement = _pluginElement;
+
+- (id)initWithPDFDocument:(RetainPtr<PDFDocument>)document andElement:(WebCore::HTMLPlugInElement*)element
+{
+    if (!(self = [super init]))
+        return nil;
+
+    _pdfDocument = document;
+    _pluginElement = element;
+    return self;
+}
+
+- (void)setPDFPluginBase:(WebKit::PDFPluginBase*)plugin
+{
+    _pdfPlugin = plugin;
+}
+
+- (void)setPDFDocument:(RetainPtr<PDFDocument>)document
+{
+    _pdfDocument = document;
+}
+
+- (BOOL)isAccessibilityElement
+{
+    return YES;
+}
+
+- (id)accessibilityFocusedUIElement
+{
+    RefPtr plugin = _pdfPlugin.get();
+    if (plugin) {
+        if (RefPtr activeAnnotation = plugin->activeAnnotation()) {
+            if (WebCore::AXObjectCache* existingCache = plugin->axObjectCache()) {
+                if (RefPtr object = existingCache->getOrCreate(activeAnnotation->element()))
```

Comment:
Put curly braces around the body of the condition, since the macros make it span multiple lines.